### DISCUSSION TITLE
How to iterate over property values using 'for' loop?

### DISCUSSION THREAD

**Discussion by wwstory** [slint-ui/slint#4285] [Q&A]
I try iterate over property in callback function, but it seems that it doesn't have such functionality. I just want to implement a feature for selecting all items in a table view. I prefer not to use Rust/C++, and it would be more reasonable for this to be implemented by the UI.

example: https://slint.dev/releases/1.3.2/docs/slint/src/recipes/recipes#custom-table-view

like this.
```
CheckBox {
    toggled => {
        // root.values[:] = self.checked;
        for r[i] in root.values {
            r.select = self.checked;
        }
    }
}
```

**Comment by ogoffart** [comment]
We are trying to prevent turning the Slint language into a programming language. We want to avoid making it possible to write the logic in the .slint file.
So you will have to do such logic in Rust or C++

**Reply by bjorn** [reply]
I tried to solve it only in Slint, in case a purely visual indication would suffice, but it does not work because toggling a checkbox destroys any binding to its `checked` property ([Slintpad](https://slint.dev/releases/1.3.2/editor/?snippet=import+%7B+VerticalBox%2C+ListView+%2C+CheckBox%7D+from+%22std-widgets.slint%22%3B%0A%0Astruct+Entry+%7B%0A++++checked%3A+bool%2C%0A++++label%3A+string%2C%0A%7D%0A%0Aexport+component+Example+inherits+Window+%7B%0A++++property+%3C%5BEntry%5D%3E+values%3A+%5B%0A++++++++%7B+label%3A+%22Foo%22%2C+checked%3A+false+%7D%2C%0A++++++++%7B+label%3A+%22Bar%22%2C+checked%3A+false+%7D%2C%0A++++++++%7B+label%3A+%22Baz%22%2C+checked%3A+false+%7D%2C%0A++++%5D%3B%0A%0A++++Rectangle+%7B%0A++++++++VerticalBox+%7B%0A++++++++++++select-all+%3A%3D+CheckBox+%7B+text%3A+%22Select+All%22%3B+%7D%0A++++++++++++Rectangle+%7B+height%3A+1px%3B+background%3A+white%3B+%7D%0A++++++++++++for+r+in+root.values+%3A+CheckBox+%7B%0A++++++++++++++++padding%3A+5px%3B%0A++++++++++++++++text%3A+r.label%3B%0A++++++++++++++++checked%3A+r.checked+%7C%7C+select-all.checked%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)). And I don't think there is currently any way to restore this binding?

It's a bit unfortunate, but in such a case, then one would indeed need to expose a logic function to set the selected state, which would probably restore the binding as a result of re-creating the items due to the change callback from the data model?

**Comment by heipiao233** [comment]
Use recursion?
